### PR TITLE
Small improvements to welcome screen (UI only)

### PR DESCRIPTION
**Pull request by isidorosp**
* Prevents the "flashing" of the UI during loading before the welcome screen is rendered by hiding the application UI until a user is logged in
* Prevents a double overlay that occurred when both the Premium Modal and the Welcome Screen overlay were both up at the same time (which caused a difference in the color of the background)
* Adds a subtle fade transition to the welcome screen (you see this after you log in or close the premium popup)

TODO:

- [x] Figure out if we want to repurpose MessageOverlay (it's not used anywhere except for the Premium popup currently)
- [x] Clean up AccountManagement.vue (split into smaller components?

### CODE REVIEW

File: electron-app/src/App.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <v-app id="rotki">
-    <div v-if="logged">
+    <div v-show="logged">
```
Comment (kelsos) [inline electron-app/src/App.vue:3]: was the flicker because we were removing the element from dom?

Comment (isidorosp) [reply electron-app/src/App.vue:3]: the If is something I added in the first commit, the flicker is caused by the fact that the full app is actually rendering (and visible) before the overlay is, this change makes sure it's hidden before the welcome screen overlay comes up.

**Comment by isidorosp** [comment]
This is a UI-only update for the welcome screen to fix a small "flash"/glitch that occurs sometimes based on how fast the app is rendering. Tests passed on my side & GH Actions but not travis.

@kelsos can you have a look and let me know what you would like me to do with the AccountManagement (split into sub-components or not) and the MessageOverlay?

**Comment by kelsos** [comment]
> This is a UI-only update for the welcome screen to fix a small "flash"/glitch that occurs sometimes based on how fast the app is rendering. Tests passed on my side & GH Actions but not Travis (OSX only).
> 
> @kelsos can you have a look and let me know what you would like me to do with the AccountManagement (split into sub-components or not) and the MessageOverlay?

Splitting into subcomponents sounds like a good idea. What do you have in mind?

**Comment by kelsos** [comment]
MessageOverlay had a couple of different things in the past (a welcome message, and the update thing) the update thing is now with the indicator and the welcome was removed early on. We can basically rename it and get all the text and functionality inside instead. If you have a better idea I am also open to that.

**Comment by isidorosp** [comment]
> > This is a UI-only update for the welcome screen to fix a small "flash"/glitch that occurs sometimes based on how fast the app is rendering. Tests passed on my side & GH Actions but not Travis (OSX only).
> > @kelsos can you have a look and let me know what you would like me to do with the AccountManagement (split into sub-components or not) and the MessageOverlay?
> 
> Splitting into subcomponents sounds like a good idea. What do you have in mind?

Nothing extravagant, just splitting it up to clean up the page a bit better (e.g. put the login card, the new user card, in their own page), but it's not really necessary. In general I think it will be good to "componentize" that card layout on top of the overlay for future use, but I'd like to work on making a list of all those things together.

**Comment by kelsos** [comment]
sounds good to me :+1:

**Comment by LefterisJP** [comment]
So what's remaining here for merging?

**Comment by kelsos** [comment]
@LefterisJP as far as I am aware @isidorosp wants to do some component splitting and cleanup before merging.

**Comment by isidorosp** [comment]
Hey @kelsos , I took another look and I don't think there's anything that we really need to pull out at least for this release. I think eventually when we sit down and design the shared components we can determine what can/should be extracted from here and/or replaced with the common component(s) we end up designing. 

In terms of separating sub-content out for readability, "sign in" and "create user" are already separate and only the Premium modal remains, but I'm not sure there's a lot of good reasons to bother splitting it out.

File: electron-app/src/components/Login.vue [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div v-if="displayed" class="rotki-welcome">
+  <div v-if="displayed" class="rotki-welcome__login">
```
Comment (kelsos) [inline electron-app/src/components/Login.vue:2]: If we follow the proper naming scheme for the components this should be just login following bem.

**Comment by isidorosp** [comment]
@kelsos ready sir!

**Pull request merged**